src/app/InteractionModelEngine.cpp

```diff
@@ -99,11 +99,6 @@ CHIP_ERROR InteractionModelEngine::Init(Messaging::ExchangeManager * apExchangeM
 void InteractionModelEngine::Shutdown()
 {
     mpExchangeMgr->GetSessionManager()->SystemLayer()->CancelTimer(ResumeSubscriptionsTimerCallback, this);
-
-    // TODO: individual object clears the entire command handler interface registry.
-    //       This may not be expected.
-    CommandHandlerInterfaceRegistry::UnregisterAllHandlers();
```

Comment:
Well, hold on.  We do in fact need to UnregisterAllHandlers() somewhere on shutdown.  Where is that happening?

Reply:
We are symmetric with AAI: there is no UnregisterAll for AAI, so i figured we would not use it for CHI right now.

We do have a UnregisterAllForEndpoint in endpoint shutdowns.

Reply:
I can keep the unregister, however because there is no ownership of this anymore, it is odd. If we run multiple IME in paralle, having any IME shutdown clear the handlers for all the others IME seems bad.

We do not do this for AAI.

Reply:
When we run multiple IME in parallel, each one will have a separate registry is the point.

For AAI, we should have a "remove all on shutdown" as well if we want to allow any sort of non-static registration, but AAI was initially created with static-only things in mind.

Reply:
I think the question is if this PR is acceptable now or should be deferred until we have a usable DataModel (or DataModel::Provider as the new name) inside the IME.

The reason I am splitting this is to have advance work to decouple IME from all implementations (so they use a provider instead of the IME). My concern is that given that CodeGen metadata is used to list available things, including Generated/Accepted commands, this is still centrally managed and code generated today to some extent.


Review feedback request on path forward:
   - keep the odd clear in here (even if ownership is odd)
   - remove the clear claiming 'things are centrally managed anyway, and since IME does not own the registry, it should not clear it on shutdown'
   
I do not care as much since I imagine medium-term we will have a real data model split anyway which should make things sane (and codegen data model being a singleton can absolutely do the clear of AAI/CHI as it wants).

Reply:
I think this PR is fine in its current general form, but I think we should keep the odd clear in here for now, and later on once we have separate registries for separate stack instances the clear would move into stack instance shutdown.

Reply:
Added clear back.